src/components/Table/CustomTable.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState, useEffect } from 'react';
+import { Table, Spin } from 'antd';
+
+import PaginationComponent from './components/Pagination';
+import TableHeader from './components/TableHeader';
+import Navbar from './components/Navbar';
+
+import './CustomTable.scss';
+
+const CustomTable = ({ data, columns, totalCount, page, pageSize, onPageChange, onPageSizeChange, isLoading, navLabels, selectedNavLabel, onNavSelect, onSearchChange }) => {
+  const [pageData, setPageData] = useState(null);
+  const [filter, setFilter] = useState(selectedNavLabel);
+
+  const handlePageSizeChange = (value) => {
+    onPageChange(1);
+    onPageSizeChange(parseInt(value));
+  };
+
+  const handlePageChange = (page) => {
+    onPageChange(page);
+  };
+
+  const handleFilterChange = (label) => {
+    setFilter(label);
+  };
+
+  useEffect(() => {
+    const filteredData = data.filter((item) => {
```

Comment:
you need to do this on backend. To be sure you get right data you will need to filter them on backend